src/java.base/share/classes/java/text/DecimalFormat.java

```diff
@@ -55,45 +55,65 @@
+ * numbers in any locale, including support for Western, Arabic, and Indic digits.
+ * It also supports different kinds of numbers, including integers (123), fixed-point
  * numbers (123.4), scientific notation (1.23E4), percentages (12%), and
- * currency amounts ($123).  All of these can be localized.
+ * currency amounts ($123).
  *
- * <p>To obtain a {@code NumberFormat} for a specific locale, including the
- * default locale, call one of {@code NumberFormat}'s factory methods, such
- * as {@code getInstance()}.  In general, do not call the
- * {@code DecimalFormat} constructors directly, since the
- * {@code NumberFormat} factory methods may return subclasses other than
- * {@code DecimalFormat}. If you need to customize the format object, do
- * something like this:
+ * <h2>Getting a DecimalFormat</h2>
  *
- * <blockquote>{@snippet lang=java :
- * NumberFormat numFormat = NumberFormat.getInstance(loc);
- * if (numFormat instanceof DecimalFormat decFormat) {
- *     decFormat.setDecimalSeparatorAlwaysShown(true);
- * }
- * }</blockquote>
+ * To get a {@code DecimalFormat} for the default Locale, use one of the {@code
+ * NumberFormat} static factory methods listed below.
+ * The following formats all provide an example of formatting the {@code Number}
+ * "2000.50" with the {@link java.util.Locale#US US} locale as the default locale.
  *
- * <p>A {@code DecimalFormat} comprises a <em>pattern</em> and a set of
- * <em>symbols</em>.  The pattern may be set directly using
- * {@code applyPattern()}, or indirectly using the API methods.  The
- * symbols are stored in a {@code DecimalFormatSymbols} object.  When using
- * the {@code NumberFormat} factory methods, the pattern and symbols are
- * read from localized {@code ResourceBundle}s.
+ * <ul>
+ * <li> Use {@link #getInstance()} or {@link #getNumberInstance()} to get
+ * a decimal format. For example, {@code "2,000.5"}.
+ * <li> Use {@link #getIntegerInstance()} to get an integer number format.
+ * For example, {@code "2,000"}.
+ * <li> Use {@link #getCurrencyInstance} to get a currency number format.
+ * For example, {@code "$2,000.50"}.
+ * <li> Use {@link #getPercentInstance} to get a format for displaying percentages.
+ * For example, {@code "200,050%"}.
+ * <li> Use one of the {@code DecimalFormat} constructors to obtain a {@code
+ * DecimalFormat} with further customization.
+ * </ul>
  *
- * <h2 id="patterns">Patterns</h2>
+ * <small>Note: It is recommended to use one of the NumberFormat factory methods
+ * which is tailored to the conventions of the given locale to retrieve a
+ * DecimalFormat.</small>
+ *<p>
+ * Alternatively, if a {@code DecimalFormat} for a different locale is required, use
+ * one of the factory method variants that take {@code locale} as a parameter,
+ * for example, {@link #getIntegerInstance(Locale)}. {@link
+ * NumberFormat#getAvailableLocales()} can be used to determine if the desired
+ * locale is supported.
```

Comment:
I don't think this section is helpful, or rather, may mislead the users. As the original text explains, factory methods in `NumberFormat` do not guarantee to return `DecimalFormat` instances by design.

Reply:
OK, that's a good point. In all 3 of these classes, I adjusted the wording to recommend using the factory methods for a locale's standard formatting. However, I made it apparent that the resultant Object can be any subclass of NumberFormat, and that if users want to cast, (so that they have access to the subclass instance methods), they should do appropriate type checking first.

All the other DecimalFormat comments you left should be implemented as well.